### ISSUE TITLE
"Warning: could not match transformation" when importing multiple modules in the same line

### ISSUE THREAD

**Issue by jakkdl** [felix-hilden/sphinx-codeautolink#142]
### Issue
In https://github.com/python-trio/trio/pull/2968 I added codeautolink to the trio documentation, and the parser failed when it stumbled upon a codeblock that started with
```python
import asyncio, trio
```
giving the warning
```
/home/h/Git/trio/docs/source/reference-lowlevel.rst: WARNING: Could not match transformation of `asyncio` on source lines 1-1
```

Changing the code to instead do
```python
import asyncio
import trio
```
got rid of the warning.

### Expected behavior
It should have been able to parse the multi-import statement.

### Steps to reproduce
I tried playing around with adding a test case to try and reproduce the error, but didn't manage to. So unless it's immediately obvious to you what the flaw might be I'll try to create an MVP starting from the other end with https://github.com/python-trio/trio/pull/2968/files/1c76b8444127112e0c5b81afd0e48d8e48d48011..43061442471a4658e4659d3d1a32e1e079a535e9

These were the settings in conf.py:
```python
codeautolink_autodoc_inject = False
codeautolink_global_preface = """
import trio
from trio import *
"""
```

versions:
sphinx-codeautolink==0.15.0
sphinx==7.1.2
python=3.11

**Comment by felix-hilden** [comment]
Thank you for the report! I'll add the case to our test suite soon 👌 Beyond a minimal example, looking at the HTML output and comparing that to a good block could help.

**Comment by felix-hilden** [comment]
Ya know what, released 😄 Prolly easier for you to test with just a new version too! And we might save a roundtrip. Feel free to reopen if something's still wrong, but a candidate fix is available in `0.15.2`

**Comment by jakkdl** [comment]
Not getting any error on the repro after updating. Thanks!
